src/test/java/com/marklogic/kafka/connect/source/DslConstraintInjectionTest.java

```diff
@@ -1,61 +1,202 @@
+        assertEquals(expectedResult, result, "The where clause should have been injected just after the closing paren of the fromSearch function");
+    }
+
+    @Test
+    void fromSPARQL() {
+        String fromSPARQLClause = "op.fromSPARQL('PREFIX ad: <http://marklogicsparql.com/addressbook#> '+'SELECT ?firstName '+'WHERE {<#5555> ad:firstName ?firstName .}','sparql',{dedup:'on',base:'http://marklogicsparql.com/id#'})";
+        String userDsl = fromSPARQLClause + ".orderBy(op.asc(" + constraintColumn + ")).limit(1000)";
+        parsedConfig.put(MarkLogicSourceConfig.DSL_QUERY, userDsl);
+        String expectedResult = fromSPARQLClause + expectedWhereClause + ".orderBy(op.asc(" + constraintColumn + ")).limit(1000)";
+
+        String result = injectValueIntoQuery();
+        assertEquals(expectedResult, result, "The where clause should have been injected just after the closing paren of the fromSearch function");
+    }
+
+    @Test
+    void fromSQL() {
+        String fromSQLClause = "op.fromSQL('SELECT employees.FirstName, employees.LastName FROM employees')";
+        String userDsl = fromSQLClause + ".orderBy(op.asc(" + constraintColumn + ")).limit(1000)";
+        parsedConfig.put(MarkLogicSourceConfig.DSL_QUERY, userDsl);
+        String expectedResult = fromSQLClause + expectedWhereClause + ".orderBy(op.asc(" + constraintColumn + ")).limit(1000)";
+
+        String result = injectValueIntoQuery();
+        assertEquals(expectedResult, result, "The where clause should have been injected just after the closing paren of the fromSearch function");
+    }
+
+    @Test
+    void fromTriples() {
+        String fromTriplesClause = "op.fromTriples([op.pattern(person,onto('birthPlace'),resource('Brooklyn')),op.pattern(person,foaf(\"name\"),op.col(\"name\"))])";
+        String userDsl = fromTriplesClause + ".orderBy(op.asc(" + constraintColumn + ")).limit(1000)";
+        parsedConfig.put(MarkLogicSourceConfig.DSL_QUERY, userDsl);
+        String expectedResult = fromTriplesClause + expectedWhereClause + ".orderBy(op.asc(" + constraintColumn + ")).limit(1000)";
+
+        String result = injectValueIntoQuery();
+        assertEquals(expectedResult, result, "The where clause should have been injected just after the closing paren of the fromSearch function");
+    }
+
+    @Test
+    void fromView() {
+        String fromViewClause = "op.fromView('main','employees',null,op.fragmentIdCol('docId'))";
+        String userDsl = fromViewClause + ".orderBy(op.asc(\"ID\")).limit(1000)";
+        parsedConfig.put(MarkLogicSourceConfig.DSL_QUERY, userDsl);
+        String expectedResult = fromViewClause + expectedWhereClause + ".orderBy(op.asc(\"" + constraintColumn + "\")).limit(1000)";
+
+        String result = injectValueIntoQuery();
+        assertEquals(expectedResult, result, "The where clause should have been injected just after the closing paren of the fromSearch function");
+    }
+
+    private String injectValueIntoQuery() {
```

Comment:
This can do a little more work and thus make every test simpler - you can change it to:

    private String injectValueIntoQuery(String query)

And then it will stuff the value into parsedConfig. And then every test can remove that line that stuffs it into parsedConfig. 

You could even make this method take care of everything:

    private void assertValueIsInjected(String userQuery, String expectedResult, String message)

And then have tests like this:

```
    @Test
    void fromView() {
        String fromViewClause = "op.fromView('main','employees',null,op.fragmentIdCol('docId'))";
        assertValueIsInjected(
            fromViewClause + ".orderBy(op.asc(\"ID\")).limit(1000)",
            fromViewClause + expectedWhereClause + ".orderBy(op.asc(\"" + constraintColumn + "\")).limit(1000)",
            "The where clause should have been injected just after the closing paren of the fromSearch function"
        );
    }
```

That might be a little too much, but I would at least move the parsedConfig bit into the private method.